hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -554,6 +564,11 @@ private CompletableFuture<Message> writeStateMachineData(
     CompletableFuture<ContainerCommandResponseProto> writeChunkFuture =
         CompletableFuture.supplyAsync(() -> {
           try {
+            try {
+              checkContainerHealthy(write.getBlockID().getContainerID());
+            } catch (StorageContainerException e) {
+              return ContainerUtils.logAndReturnError(LOG, e, requestProto);
```

Comment:
We should follow same exception approach of catch and throw exception as done below for other failure. This need not follow applyTransaction way of handling, may have some impact.

Reply:
As discuss over last meeting, need have similar exception handling for new and existing cases.

Reply:
done